.github/actions/setup-node/action.yml

```diff
@@ -47,16 +47,21 @@ runs:
           (steps.node-modules-restore.outputs.cache-hit == 'true') && 'true' || ''
         }}' >> "$GITHUB_ENV"
 
-    - name: Enable corepack
-      shell: bash
-      run: corepack enable
+    - name: Install pnpm
+      uses: pnpm/action-setup@v2.4.0
+      with:
```

Comment:
why this is required?

Reply:
Setup node uses pnpm but the corepack is not enabled yet so the binary is missing.

Reply:
then enable corepack instead

Reply:
It is installed with the node setup

Reply:
sure, but corepack is also available if nodejs is already available

Reply:
nodejs is not available with the act images I tested

Reply:
😕, with this change we need a new regex manager to update the pnpm version

Reply:
Not sure it worth the efforts. This section will be removed as soon as this PR is merged:
https://github.com/actions/setup-node/pull/651

Reply:
Do we want to merge this before then? That PR is conflicted and waiting for someone to resume it and then hopefully the maintainers to merge it

Reply:
I do think it has value in merging it. Waiting for a general approval and then I'll rebase.

Reply:
We'll need a regex manager added to keep it updated though. It's already out of date :)

Reply:
Done

Reply:
I would prefer to leave the corepack enable when not on act. maybe we should build our own act image to not need the huge default act image.